src/components/app/ServiceWorkerManager.js

```diff
@@ -82,56 +82,61 @@ class ServiceWorkerManagerImpl extends PureComponent<Props, State> {
-        );
-      },
-      onUpdateReady: () => {
-        console.log(
-          '[ServiceWorker] We have downloaded the new assets and we are ready to go.'
-        );
-        this.setState({
-          installStatus: 'pending',
-          isNoticeDisplayed: true,
-        });
-      },
-      onUpdated: () => {
-        // The update could have been triggered by this tab or another tab.
-        // We distinguish these cases by looking at this.state.installStatus.
-        console.log(
-          '[ServiceWorker] The new version of the application has been enabled.'
-        );
+    if (!('serviceWorker' in navigator)) {
+      return;
+    }
 
-        if (this.state.installStatus === 'installing') {
-          // In this page the user clicked on the "reload" button.
-          this.reloadPage();
-          return;
-        }
-
-        // In another page, the user clicked on the "reload" button.
-
-        const ready =
-          !this._hasDataSourceProfile() || this._isProfileLoadedAndReady();
-
-        this.setState({
-          installStatus: 'installed',
-          // But if we weren't quite ready, we should write it in the notice.
-          updatedWhileNotReady: !ready,
-        });
-      },
-      onUpdateFailed: () => {
-        console.log(
-          '[ServiceWorker] We failed to update the application for an unknown reason.'
-        );
-      },
+    const wb = (this._workbox = new Workbox('/sw.js'));
+    wb.register();
+
+    wb.addEventListener('installing', () => {
+      console.log(
+        '[ServiceWorker] An update has been found and the browser is downloading the new assets.'
+      );
+    });
+
+    wb.addEventListener('installed', () => {
+      // We cached all assets.
+      console.log('[ServiceWorker] App is ready for offline usage!');
+    });
+
+    wb.addEventListener('waiting', () => {
+      // Update is ready to be applied.
+      console.log(
+        '[ServiceWorker] We have downloaded the new assets and we are ready to go.'
+      );
+      this.setState({
+        installStatus: 'pending',
+        isNoticeDisplayed: true,
+      });
+    });
+
+    wb.addEventListener('activated', () => {
+      // The update could have been triggered by this tab or another tab.
+      // We distinguish these cases by looking at this.state.installStatus.
+      console.log(
+        '[ServiceWorker] The new version of the application has been enabled.'
+      );
+
+      if (this.state.installStatus === 'installing') {
+        // In this page the user clicked on the "reload" button.
+        this.reloadPage();
+        return;
+      }
+
+      // In another page, the user clicked on the "reload" button.
+
+      const ready =
+        !this._hasDataSourceProfile() || this._isProfileLoadedAndReady();
+
+      this.setState({
+        installStatus: 'installed',
+        // But if we weren't quite ready, we should write it in the notice.
+        updatedWhileNotReady: !ready,
+      });
```

Comment:
I didn't change the logic at all, just replaced the event handlers by the new event handlers.